src/main/kotlin/net/leanix/githubagent/services/WebhookEventService.kt

```diff
@@ -51,15 +51,17 @@ class WebhookEventService(
 
         val installationToken = getInstallationToken(pushEventPayload.installation.id)
 
-        if (pushEventPayload.ref == "refs/heads/$defaultBranch") {
-            handleManifestFileChanges(
-                defaultBranch,
-                headCommit,
-                repositoryFullName,
-                owner,
-                repositoryName,
-                installationToken
-            )
+        headCommit?.let {
```

Comment:
`let` does not make sense here. 
it should just be `if (headCommit != null && pushEventPayload.ref == "refs/heads/$defaultBranch")`